### DISCUSSION TITLE
Using pybind11 with pocketpy interpretor

### DISCUSSION THREAD

**Discussion by rhvarrier** [pybind/pybind11#4766] [Q&A]
I've built a C++ library that exposes it's API using pybind11 and I'm able to use the library functions from python under Linux. Howeverre, while building the library pybind11 needs to link against a CPython, which is not always possible in some of my build environements (as CPython is not supported by the compiler I'm using). 
I came across [pockectpy](https://github.com/blueloveTH/pocketpy), which is a zero dependecy single file Python interpreter. 
Can pybind11 be used with this library instead of CPython, which would allow me to have a fully standalone library with integrated Python interpretor.
I know that pocketpy does not expose all python functionalities but the supported feature are more than enough for my use case

**Comment by allsey87** [comment]
There is an option to build a Pybind11 module independent of the interpreter, see [`NOPYTHON` mode](https://pybind11.readthedocs.io/en/stable/compiling.html#advanced-nopython-mode). Whether or not that will be compatible with pocketpy depends on that project and whether it uses the same APIs and semantics as CPython.

**Comment by UnidayStudio** [comment]
Did anyone had any luck with this? I haven't tried it yet, but I would love to replace python with pocketpy in the engine we use here in our studio.

**Reply by rhvarrier** [reply]
Thanks for your response. I'm in a very similar case, as I'm using pybind11 to add scripting to an application and having pocketpy support in pybind11 will make make my application completely standalone.
